crates/router/src/core/errors/api_error_response.rs

```diff
@@ -426,6 +428,9 @@ impl common_utils::errors::ErrorSwitch<api_models::errors::types::ApiErrorRespon
             Self::AddressNotFound => {
                 AER::NotFound(ApiError::new("HE", 4, "Address does not exist in our records", None))
             },
+            Self::GenericNotFoundError { message } => {
+                AER::NotFound(ApiError::new("HE", 5, format!("{message:?}"), None))
+            },
```

Comment:
I'm confused, is the error code supposed to be `IR_19` or `HE_05`?

Also, message is a `String`, you should be able to use only `format!("{message}")`?

Reply:
We just want to throw a descriptive error that's why put a String any suggestions ?

Reply:
`String` should be fine, I just meant to say that you don't need the `Debug` implementation of `message`.

Reply:
Makes sense